torch/csrc/distributed/rpc/init.cpp

```diff
@@ -409,31 +410,33 @@ If the future completes with an error, an exception is thrown.
          const py::object& userCallable,
          const py::tuple& argsTuple,
          const py::dict& kwargsDict) {
-        DCHECK(PyGILState_Check());
+        DCHECK(!PyGILState_Check());
         // No need to catch exception here, if function can not be found,
         // exception will be thrown in get_function() call; if args do not match
         // with function schema, exception will be thrown in
         // createStackForSchema() call.
-        auto qualifiedNameStr = c10::QualifiedName(
-            py::cast<std::string>(py::module::import("torch.jit")
-                                      .attr("_qualified_name")(userCallable)));
-        auto qualifiedName = c10::QualifiedName(qualifiedNameStr);
-        auto functionSchema = PythonRpcHandler::getInstance()
-                                  .jitCompilationUnit()
-                                  ->get_function(qualifiedName)
-                                  .getSchema();
-        Stack stack = torch::jit::createStackForSchema(
-            functionSchema,
-            argsTuple.cast<py::args>(),
-            kwargsDict.cast<py::kwargs>(),
-            c10::nullopt);
-        py::gil_scoped_release release;
+        auto& pythonRpcHandler = PythonRpcHandler::getInstance();
+        c10::QualifiedName qualifiedName =
+            pythonRpcHandler.getQualifiedName(userCallable);
+        c10::FunctionSchema functionSchema =
+            pythonRpcHandler.jitCompilationUnit()
+                ->get_function(qualifiedName)
+                .getSchema();
+        Stack stack;
+        {
+          py::gil_scoped_acquire acquire;
+          stack = torch::jit::createStackForSchema(
+              functionSchema,
+              argsTuple.cast<py::args>(),
+              kwargsDict.cast<py::kwargs>(),
+              c10::nullopt);
+        }
         DCHECK(!PyGILState_Check());
```

Comment:
This is redundant?

Reply:
This is to revert back to https://github.com/pytorch/pytorch/pull/33610

This macro does not produce code unless in debug mode (gcc -g).

Not plan to change it.